sdk/core/azure_core_amqp/src/connection.rs

```diff
@@ -0,0 +1,315 @@
+// Copyright (c) Microsoft Corp. All Rights Reserved.
+// cspell: words amqp sasl
+
+use super::value::{AmqpOrderedMap, AmqpSymbol, AmqpValue};
+use azure_core::error::Result;
+use std::fmt::Debug;
+use time::Duration;
+use url::Url;
+
+pub struct AmqpConnectionOptions {
+    pub(crate) max_frame_size: Option<u32>,
+    pub(crate) channel_max: Option<u16>,
+    pub(crate) idle_timeout: Option<Duration>,
+    pub(crate) outgoing_locales: Option<Vec<String>>,
+    pub(crate) incoming_locales: Option<Vec<String>>,
+    pub(crate) offered_capabilities: Option<Vec<AmqpSymbol>>,
+    pub(crate) desired_capabilities: Option<Vec<AmqpSymbol>>,
+    pub(crate) properties: Option<AmqpOrderedMap<AmqpSymbol, AmqpValue>>,
+    pub(crate) buffer_size: Option<usize>,
+}
+
+impl AmqpConnectionOptions {
+    pub fn builder() -> builders::AmqpConnectionOptionsBuilder {
+        builders::AmqpConnectionOptionsBuilder::new()
+    }
+}
+
+pub trait AmqpConnectionTrait {
```

Comment:
Not common to have "Trait" in a trait. It's obvious what it is.

Reply:
Yeah, I know - the problem here is that I have three related types:
1) The type `AmqpConnection`
1) The implementation of the type `AmqpConnectionImpl`
1) The trait which defines the API surface for the type `AmqpConnectionTrait`.

I have been struggling with what to call these three - I want the concrete type to be named `AmqpConnection`. I don't care about #2 and #3, but they each need their own name. And thus my dilemma.

Reply:
Is a trait even necessary and does it need to be exposed publicly?

We can run with it for now, though. Curious if @demoray, @eholk, @yoshuawuyts, or others know of some precedent or have ideas here. Not blocking, though.

Reply:
I *think* the trait is necessary, and it unfortunately does have to be public - I've tried having it be private and I get a bunch of compiler errors indicating that the compiler can't generate correct code without visibility into the connection :(.